executor/aggregate_test.go

```diff
@@ -1297,3 +1298,59 @@ func (s *testSuiteAgg) TestIssue20658(c *C) {
 		}
 	}
 }
+
+func (s *testSuiteAgg) TestIssue15284(c *C) {
```

Comment:
We can also modify variables tidb_max_chunk_size to simulate that the executor call Next() many times.